src/Agent/NewRelic/Agent/Core/Utilization/VendorInfo.cs

```diff
@@ -273,50 +274,83 @@ private string GetProcessEnvironmentVariable(string variableName)
         private IVendorModel GetDockerVendorInfo()
         {
 #if NETSTANDARD2_0
-			bool isLinux = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
-			int subsystemsIndex = 1;
-			int controlGroupIndex = 2;
-
-			if (isLinux)
-			{
-				try
-				{
-					string id = null;
-					var fileLines = File.ReadAllLines("/proc/self/cgroup");
-
-					foreach(var line in fileLines)
-					{
-						var elements = line.Split(StringSeparators.Colon);
-						var cpuSubsystem = elements[subsystemsIndex].Split(StringSeparators.Comma).FirstOrDefault(subsystem => subsystem == "cpu");
-						if (cpuSubsystem != null)
-						{
-							var controlGroup = elements[controlGroupIndex];
-							var match = Regex.Match(controlGroup, ContainerIdRegex);
-							
-							if (match.Success)
-							{
-								id = match.Value;
-							}
-						}
-					}
-
-					if(id == null)
-					{
-						return null;
-					}
-
-					return new DockerVendorModel(id);
-				}
-				catch
-				{
-					return null;
-				}
-				
-			}
+            bool isLinux = RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+
+            if (isLinux)
+            {
+                try
+                {
+                    var fileContent = File.ReadAllText("/proc/self/mountinfo");
+                    var vendorModel = TryGetDockerCGroupV2(fileContent)
+                                      ?? TryGetDockerCGroupV1();
+
+                    return vendorModel;
+                }
+                catch
+                {
+                    return null;
```

Comment:
Maybe we could add a Finest-level log line here? I imagine this would be tough to troubleshoot otherwise if something goes wrong

Reply:
Oh wait, does this always execute? Maybe just log if we find one of the files but fail to parse them?

Reply:
Actually, I need to rework the logic a bit - it won't fall back to calling the V1 method if `/proc/self/mountinfo` doesn't exist.

Reply:
@chynesNR How does my latest change look?